File: Computer Programming/C++/Hashing/Sum of Unique Elements.cpp
class Solution {
public:
    int sumOfUnique(vector<int>& nums) 
    {
        unordered_map<int,int> hash;
        for(auto it : nums)
        {
            hash[it] ++;
        }    
        int total = 0;
        for(auto it : nums)
        {
            if(hash[it] == 1)
            {
                total += it;
            }
        }

        return total;
    }
};

File: Computer Programming/Python/Hashing/Sum of Unique Elements.py
class Solution(object):
    def sumOfUnique(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        
        mp = {}
        n = len(nums)

        for i in range(n):
            if nums[i] in mp:
                mp[nums[i]] += 1

            else:
                mp[nums[i]] = 1
        total = 0
        for i in range(n):
            if mp[nums[i]] == 1:
                total += nums[i]

        return total
